synapse/rest/client/v1/login.py

```diff
@@ -39,6 +40,7 @@
 
 class LoginRestServlet(RestServlet):
     PATTERNS = client_patterns("/login$", v1=True)
+    APPSERVICE_TYPE = "uk.half-shot.msc2778.login.application_service"
```

Comment:
Shouldn't this be a tuple of potential types? I see MSC2778 is referenced, which proposes `m.login.application_service` as its appservice_type, that means that this custom event type, and the new proposed type, at some point in the future, are both valid types.

```suggestion
    APPSERVICE_TYPES = ("uk.half-shot.msc2778.login.application_service",)
```

Reply:
They should probably just be separate variables like `JWT_TYPE` and `JWT_TYPE_DEPRECATED`

Reply:
I think we'd want to do it like JWT, just for consistency. (Or we can make JWT a tuple, but either way we should be consistent.)

Reply:
Can you move this below the standardized ones? These aren't actually in alphabetical order.